libraries/debug-certificate-manager/src/CertificateManager.ts

```
@@ -342,6 +372,49 @@ export class CertificateManager {
     }
   }
 
+  /**
+   * Get the expiration dates of the stored CA and server certificates.
+   * Returns undefined for certificates that don't exist or cannot be parsed.
+   *
+   * @public
+   */
+  public async getCertificateExpirationAsync(): Promise<ICertificateExpiration> {
+    const { certificateData, caCertificateData } = this._certificateStore;
+
+    let caCertificateExpiration: Date | undefined;
+    let serverCertificateExpiration: Date | undefined;
+
+    try {
+      const forge: typeof import('node-forge') = await import('node-forge');
+
+      if (caCertificateData) {
+        try {
+          const caCertificate: pki.Certificate = forge.pki.certificateFromPem(caCertificateData);
+          caCertificateExpiration = caCertificate.validity.notAfter;
+        } catch (error) {
+          // no-op
```

We're just swallowing all errors here?

---

I'll add logging to the try/catch